packages/grid/_modules_/grid/models/colDef/gridBooleanColDef.tsx

```diff
@@ -4,6 +4,14 @@ import { renderBooleanCell } from '../../components/cell/GridBooleanCell';
 import { renderEditBooleanCell } from '../../components/cell/GridEditBooleanCell';
 import { gridNumberComparer } from '../../utils/sortingUtils';
 import { getGridBooleanOperators } from './gridBooleanOperators';
+import { GridValueFormatterParams } from '../params/gridCellParams';
+
+function gridBooleanFormatter({ value }: GridValueFormatterParams) {
+  if (typeof value === 'boolean') {
+    return value.toString();
```

Comment:
This value should be translated. We already have `booleanCellTrueLabel` and `booleanCellFalseLabel` that are used to set the title of the icon.

https://github.com/mui-org/material-ui-x/blob/363a7a1cff3f4cb1dba20bb17264acf7d80d89f6/packages/grid/_modules_/grid/components/cell/GridBooleanCell.tsx#L31

Reply:
Is it a separate thing? the `GridBooleanCell` works as expected if no `renderCell` is provided, this fix is for custom `renderCell`.

Reply:
`GridBooleanCell` is not working quite well. When the data is exported to CSV, the `renderCell` is not rendered but the formatted value is used instead. Since there was no default formatter before this PR, it would cast the boolean to string, generating "true" or "false". My point is that if we're going to add a default value formatter, it should translate the resulting value according to the user's locale. That being said, the value should be "True" or "False", when the English locale is used. Also, we don't need to check if `typeof value` is boolean, any value that evaluates to true is fine.

Reply:
Okay, that's a good insight.

Reply:
If it's simpler. We can undo the change of the value formater. Fixing renderCell to correctly process react nodes is already great.